docs/api/specs/webConnectionProtocol.md

```diff
@@ -0,0 +1,384 @@
+          }
+        }, 
+        timeout
+      );
+      // listen for the fdc3Ready event
+      window.addEventListener("fdc3Ready", listener, { once: true });
+    }
+  });
+};
+```
+
+To discover a Desktop Agent Proxy interface, locate all candidates for a `parent` window or frame by first checking the values of `window.opener` and `window.parent`. If `window.opener !== null` or `window.parent !== window` then they are candidates for a parent window or frame. As iframes can be nested we can also search for candidates that are parents of a parent frame, e.g.:
+
+```ts
+const discoverProxyCandidates = (): WindowProxy[] => {
+  const candidates = [];
+  //parent window
+  if (!!window.opener) { candidates.push(window.opener); }
+
+  //parent frames
+  let currentWin = window;
+  while (currentWin.parent !== currentWin) {
+    candidates.push(currentWin.parent);
+    currentWin = currentWin.parent;
+  }
+
+  //parent window of top-level parent frame
+  if (window !== currentWin && !!currentWin.opener) { 
+    candidates.push(currentWin.opener); 
+  }
+  return candidates;
+}
+```
+
+Setup a timer for specified timeout, and then for each `candidate` found, attempt to establish communication with it as follows:
+
+  1. Add a listener (`candidate.addEventListener("message", (event) => {})`) to receive response messages from the `candidate`.
+  2. Send a [`"WCP1Hello"`](https://fdc3.finos.org/schemas/next/api/WCP1Hello.schema.json) message to it via `postMessage`.
+
+  ```ts
+  const hello = {
+    type: "WCP1Hello",
+    payload: {
+      identityUrl: identityUrl,
+      actualUrl: actualUrl,
+      fdc3Version: "2.2",
+      intentResolver: true,
+      channelSelector: true
+    },
+    meta: {
+      connectionAttemptUuid: "bc96f1db-9b2b-465f-aab3-3870dc07b072",
+      timestamp: "2024-09-09T11:44:39+00:00"
+    }
+  };
+  candidate.postMessage(hello, { targetOrigin: "*" });
+  ```
+
+  Note that the `targetOrigin` is set to `*` as the origin of the Desktop Agent is not known at this point.
+  3. Accept the first correct response received from a candidate. Correct responses MUST correspond to either the [`WCP2LoadUrl`](https://fdc3.finos.org/schemas/next/api/WCP2LoadUrl.schema.json) or [`WCP3Handshake`](https://fdc3.finos.org/schemas/next/api/WCP3Handshake.schema.json) message schemas and MUST quote the same `meta.connectionAttemptUuid` value provided in the original `WCP1Hello` message. Stop the timeout when a correct response is received. If no response is received from any candidate, the `getAgent()` implementation MAY retry sending the `WCP1Hello` message periodically until the timeout is reached.
+  4. If a [`WCP3Handshake`](https://fdc3.finos.org/schemas/next/api/WCP3Handshake.schema.json) was received in the previous step, skip this this step and move on to 5. However, If a [`WCP2LoadUrl`](https://fdc3.finos.org/schemas/next/api/WCP2LoadUrl.schema.json) was received in the previous step:
```

Comment:
"move on to 5" should be "move on to step 5"

Reply:
fixed in 5e8ba8b01ab5ecf86081172ff87369b1faf28540